Source/Serilog.Exceptions/Destructurers/ReflectionBasedDestructurer.cs

```diff
@@ -329,6 +329,11 @@ private ReflectionInfo GenerateReflectionInfoForType(Type valueType)
             {
                 try
                 {
+                    if (values.ContainsKey(property.Name))

```

Comment:
Can you explain this change a little? I think it's just making sure that no duplicates are added. Is that something that can happen?

Perhaps a unit test would also help explain and test it.

Reply:
I agree with @RehanSaeed, I reviewed the code and since we iterate over properties name generated by reflection there is no possibility of duplicates.
I would suggest removing the code unless you can demonstrate in the unit test that there is a possibility to introduce duplicates here.

Reply:
Yeah, I should have made this a separate PR...

There is a possibility of having the same property name twice if the property is [hidden](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/new-modifier).

The first commit was this check and the [test ](https://github.com/RehanSaeed/Serilog.Exceptions/pull/117/commits/018e3ee10a7dc522f2a39beef56e5919631336a2#diff-f90b0d514a942b97d04d884f6bc71741R353) as well.

Reply:
Good spot on this Joel! If this was a separate PR, I think we could have merged now but no worries!

I'm not sure if we should ignore the property if we find a duplicate. We should probably add something to the end of the name and add it, so we don't lost anything. Thoughts @krajek?

Reply:
I had the same thoughts yesterday.
For sure, dropping one of the properties is not the way to go.

We already had similar issue with `Type` property name.

```
if (propertiesBag.ContainsProperty("Type"))
            {
                if (!propertiesBag.ContainsProperty("$Type"))
                {
                    propertiesBag.AddProperty("$Type", valueType.FullName);
                }
                else
                {
                    // If both "Type" and "$Type" are present
                    // we just give up appending exception type
                }
            }
            else
            {
                propertiesBag.AddProperty("Type", valueType.FullName);
            }
```

Reply:
In theory, there could be more than one property like this, so we should append something random to the end of the property name.

Reply:
> If this was a separate PR, I think we could have merged now but no worries!

It would still have to wait on the separate PR 😜, I noticed it because of [this](https://github.com/aspnet/EntityFrameworkCore/blob/master/src/EFCore/ChangeTracking/EntityEntry%60.cs#L46)

---
What do you guys think of prepending the `Namespace.TypeName` if there is a collision?
```csharp
private class ReflectionPropertyInfo
{
    public string Name { get; set; }
    public Type DeclaringType { get; set; }
    public Func<object, object> Getter { get; set; }
}
...
var key = values.ContainsKey(property.Name) ? $"{property.DeclaringType.FullName}.{property.Name}" : property.Name;
values.Add(key, destructuredValue);
```

Reply:
That sounds like a great idea.

Reply:
Done.